File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
       map<int,int>duplicate;
       for(int num: nums){
         duplicate[num]++;
       }
       for(auto& duplic: duplicate){
        if(duplic.second>1)
            return duplic.first;
       }
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int l=0;
        int m= nums.length -1;
        while(l<=m){
            int mid = (l+m)/2;
            if(nums[mid]>target){
                m= mid-1;
            }
            else if(nums[mid]==target)
                return mid;
            else{
                l=mid+1;
            }
        }
        return -1;
    }
}
